.github/workflows/tests.yml

```diff
@@ -0,0 +1,95 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  BATS_VERSION: v1.3.0
+
+jobs:
+  wsl1:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-2019
+        distribution:
+          - Ubuntu-20.04
+          - Ubuntu-18.04
+          # - Alpine
+          # - Debian
+          # - kali-linux
+          # - openSUSE-Leap-15.2
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Set git to use LF
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup WSL & install test dependencies
+        uses: Vampire/setup-wsl@v1
+        with:
+          distribution: ${{ matrix.distribution }}
+          additional-packages: curl git fish
+
+      - name: Install bats
+        shell: wsl-bash {0}
+        run: |
+          git clone --depth 1 --branch "${{ env.BATS_VERSION }}" https://github.com/bats-core/bats-core.git $HOME/bats-core
+          cd $HOME/bats-core
+          ./install.sh $HOME
+          $HOME/bin/bats --version
+
+      - name: Run tests
+        shell: wsl-bash {0}
+        run: |
+          $HOME/bin/bats test
+        env:
+          GITHUB_API_TOKEN: ${{ github.token }}
+
+  nix:
```

Comment:
Why is this job named `nix`?

Reply:
It is the job that runs our Bats tests on the *nix OSs. `*` is invalid in the job name when I tried naming it `*nix`. Didn't think the job name was particularly important.

The other job is called `WSL1` because though it is a Windows OS, the environment we want to test in is different distros on WSL1 environment.

Reply:
We can rename the job to whatever, I just thought this was the most descriptive at the time.